scripts/Connector/Connector_Molex/conn_molex_micro-fit-3.0_tht_top_dual_row.py

```diff
@@ -0,0 +1,275 @@
+#!/usr/bin/env python
+
+'''
+kicad-footprint-generator is free software: you can redistribute it and/or
+modify it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+kicad-footprint-generator is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with kicad-footprint-generator. If not, see < http://www.gnu.org/licenses/ >.
+'''
+
+import sys
+import os
+#sys.path.append(os.path.join(sys.path[0],"..","..","kicad_mod")) # load kicad_mod path
+
+# export PYTHONPATH="${PYTHONPATH}<path to kicad-footprint-generator directory>"
+sys.path.append(os.path.join(sys.path[0], "..", "..", ".."))  # load parent path of KicadModTree
+from math import sqrt
+import argparse
+import yaml
+from helpers import *
+from KicadModTree import *
+
+sys.path.append(os.path.join(sys.path[0], "..", "..", "tools"))  # load parent path of tools
+from footprint_text_fields import addTextFields
+
+series = "Micro-Fit_3.0"
+series_long = 'Micro-Fit 3.0 Connector System'
+manufacturer = 'Molex'
+orientation = 'V'
+number_of_rows = 2
+
+variant_params = {
+    'solder_mounting':{
+        'mount_pins': 'solder', # remove this
+        'datasheet': 'http://www.molex.com/pdm_docs/sd/430450212_sd.pdf',
+        'C_minus_B': 6,
+        'part_code': "43045-{n:02}12",
+        'alternative_codes': [
+            "43045-{n:02}13",
+            "43045-{n:02}24"
+            ]
+        },
+}
+
+pins_per_row_range = range(1,13)
+pitch = 3.0
+drill = 1.0
+pad_size = [1.6, 1.6]
```

Comment:
For pad size calculation have a look at other THT footprints.
(In them i claculate the pad size via the pad to pad clearance and minimum/maximum annular ring requirements.)

For single row:
```
pad_size = [pitch - pad_to_pad_clearance, drill + 2*max_annular_ring]
if pad_size[0] - drill < 2*min_annular_ring:
    pad_size[0] = drill + 2*min_annular_ring
if pad_size[0] - drill > 2*max_annular_ring:
    pad_size[0] = drill + 2*max_annular_ring

if pad_size[1] - drill < 2*min_annular_ring:
    pad_size[1] = drill + 2*min_annular_ring
if pad_size[1] - drill > 2*max_annular_ring:
    pad_size[1] = drill + 2*max_annular_ring

pad_shape=Pad.SHAPE_OVAL
if pad_size[1] == pad_size[0]:
    pad_shape=Pad.SHAPE_CIRCLE
```
for dual row the first row should be: (in your case pitch_row = pitch)
`pad_size = [pitch - pad_to_pad_clearance,  pitch_row - pad_to_pad_clearance]`

Reply:
Thanks for the detailed explanation! I adopted it to your suggestion.

http://www.molex.com/pdm_docs/ps/PS-43045.pdf lists voltages of up to 600 V. So I went for 1.5 mm pad clearance.